editoast/database/src/db_connection_pool.rs

```diff
@@ -21,16 +21,109 @@ use futures_util::FutureExt as _;
 use openssl::ssl::SslConnector;
 use openssl::ssl::SslMethod;
 use openssl::ssl::SslVerifyMode;
+use tokio::sync::OwnedRwLockWriteGuard;
+use tokio::sync::RwLock;
 use tracing::trace;
 use url::Url;
 
-use tokio::sync::OwnedRwLockWriteGuard;
-use tokio::sync::RwLock;
+#[cfg(any(test, feature = "testing"))]
+use test_helpers::*;
 
 use crate::DatabaseError;
 
 pub type DbConnectionConfig = AsyncDieselConnectionManager<AsyncPgConnection>;
 
+#[cfg(any(test, feature = "testing"))]
+mod test_helpers {
```

Comment:
nit: I don't believe the module is necessary, just gating the top-level definitions is enough to me. Feel free to resolve